backend/src/sockets/handler.ts

```diff
@@ -67,6 +80,11 @@ class RoomSocketHandler {
     return Promise.resolve();
   }
 
+  private getAllUsernames(): Promise<void> {
```

Comment:
looks like this function isn't doing anything?

Reply:
It emits an event

Reply:
ah didn't see that line for some reason but the function doesn't need to return a Promise

Reply:
I think it does actually to satisfy the type constraint. I made the functions in the event map have to return promises to allow async functionality

Reply:
but there's no async functionality here?

Reply:
Yes and the same is true of `createUsername`. The problem is the type of EventHandler: 
```
type EventHandler = {
  [event in Event]?: (args: any) => Promise<void>;
};
```
So we can't just change this method to not return a promise. We could put the `Promise.resolve` in the `createHandler` method but I don't see how that's better and it would break the symmetry with other methods in the class.